src/main/java/org/usfirst/frc/team1294/robot/commands/TwoPIDControllerCommand.java

```diff
@@ -0,0 +1,178 @@
+  private TurnPIDOutput turnOutput;
+  private MovePIDOutput moveOutput;
+
+  public TwoPIDControllerCommand(double heading) {
+    this(String.format("Turn to a heading of %f", heading), heading, 0, 0, 15);
+  }
+
+  public TwoPIDControllerCommand(double speed, double distance) {
+    this(String.format("Drive the current heading at a speed of %f for a distance of %f.", speed, distance), -1, speed, distance, 15);
+  }
+
+  public TwoPIDControllerCommand(double heading, double speed, double distance) {
+    this(String.format("Drive a heading of %f at a speed of %f for a distance of %f.", heading, speed, distance), heading, speed, distance, 15);
+  }
+
+  public TwoPIDControllerCommand(double heading, double speed, double distance, double timeout) {
+    this(String.format("Drive a heading of %f at a speed of %f for a distance of %f with a timeout of %f.", heading, speed, distance, timeout), heading, speed, distance, timeout);
+  }
+
+  private TwoPIDControllerCommand(String name, double heading, double speed, double distance, double timeout) {
+    requires(Robot.driveBase);
+
+    this.heading = heading;
+    this.speed = speed;
+    this.distance = distance;
+
+    turnOutput = new TurnPIDOutput();
+    moveOutput = new MovePIDOutput();
+    turnController = new PIDController(TURNPID_P, TURNPID_I, TURNPID_D, new GyroPIDSource(), turnOutput);
+    moveController = new PIDController(MOVEPID_P, MOVEPID_I, MOVEPID_D, new AverageEncoderPIDSource(), moveOutput);
+
+    LiveWindow.addActuator("TwoPIDControllerCommand", "turnController", turnController);
+    LiveWindow.addActuator("TwoPIDControllerCommand", "moveController", moveController);
+
+    // since this is a closed loop command, and may be running without operator intervention, set a timeout
+    setTimeout(timeout);
+  }
+
+  @Override
+  protected void initialize() {
+    // if heading was not provided, use current heading
+    if (heading < 0) {
+      turnController.setSetpoint(Robot.driveBase.getNormalizedAngle());
+      System.out.println("setpoint to current heading " + Robot.driveBase.getNormalizedAngle());
+    } else {
+      turnController.setSetpoint(heading);
+    }
+    turnController.setInputRange(0, 360);
+    turnController.setOutputRange(-0.4, 0.4);
+    turnController.setContinuous();
+    turnController.setAbsoluteTolerance(PID_TOLERANCE);
+    turnController.enable();
+
+    moveController.setSetpoint(Robot.driveBase.getAverageEncoderDistance() + distance);
+    moveController.setOutputRange(-speed, speed);
+    moveController.setAbsoluteTolerance(PID_TOLERANCE);
+    moveController.enable();
+
+    Robot.driveBase.setTalonsToClosedLoopSpeed();
+
+    SmartDashboard.putData(this);
+  }
+
+  @Override
+  protected void execute() {
+    Robot.driveBase.arcadeDrive(moveOutput.getOutput(), turnOutput.getOutput());
+  }
+
+  @Override
+  protected boolean isFinished() {
+    return isTimedOut() || (turnController.onTarget() && moveController.onTarget());
```

Comment:
`PIDController.onTarget()` returns `true` when the error is within the provided tollerance (set via `PIDController.setAbsoluteTollerance()`).